gen3_tracker/meta/dataframer.py

```diff
@@ -498,7 +473,11 @@ def select_coding(self, resource):
             # Handle case where no coding entries exist
             return None
 
-    def flattened_observations(self) -> Generator[dict, None, None]:
+    def flattened_observations(self, focus_type: str = None) -> Generator[dict, None, None]:
+        """flattens all observations, augmenting it with fields from .subject and .focus
+        use focus_type if you want to subset on a particular resource, eg Specimen"""
+
+        # TODO: not used, can it be deleted?
```

Comment:
In the current dataset the FHIR codings are formed in such a way where this function is not needed.

There exists valid FHIR that will not dataframe into a column title that guppy will be able to read and Guppy will crash. There needs to be something that checks to see if the codings are parsable in guppy, and if not invokes this translater to make them parsable

Reply:
will leave in a class and push up

Reply:
updated, see [normalize_for_guppy](https://github.com/ACED-IDP/gen3_util/blob/e0ca43f3268102308a82d6022ac6e5deb97f192a/gen3_tracker/meta/entities.py/#L132-L152)

Reply:
confused.
<img width="249" alt="image" src="https://github.com/user-attachments/assets/39d8471b-61e2-44b3-8e39-50a1a60439ed">

Reply:
it's not currently being used since it's not needed now. Matthew and I had an offline discussion that it can be referenced if needed for future datasets so just keeping it as util function is enough. Happy to discuss!